plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/tasks/LoadReportersTask.kt

```diff
@@ -0,0 +1,89 @@
+package org.jlleitschuh.gradle.ktlint.tasks
+
+import net.swiftzer.semver.SemVer
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.workers.WorkerExecutor
+import org.jlleitschuh.gradle.ktlint.checkMinimalSupportedKtLintVersion
+import org.jlleitschuh.gradle.ktlint.intermediateResultsBuildDir
+import org.jlleitschuh.gradle.ktlint.reporter.CustomReporter
+import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
+import org.jlleitschuh.gradle.ktlint.worker.LoadReportersWorkAction
+import javax.inject.Inject
+
+@Suppress("UnstableApiUsage")
+@CacheableTask
+internal abstract class LoadReportersTask @Inject constructor(
+    private val workerExecutor: WorkerExecutor,
+    objectFactory: ObjectFactory,
+    projectLayout: ProjectLayout,
+) : DefaultTask() {
+
+    @get:Classpath
+    internal abstract val ktLintClasspath: ConfigurableFileCollection
+
+    @get:Classpath
+    internal abstract val reportersClasspath: ConfigurableFileCollection
+
+    @get:Input
+    internal abstract val debug: Property<Boolean>
+
+    @get:Input
+    internal abstract val ktLintVersion: Property<String>
+
+    @get:Input
+    internal abstract val enabledReporters: SetProperty<ReporterType>
+
+    @get:Input
+    internal abstract val customReporters: SetProperty<CustomReporter>
+
+    @get:OutputFile
+    internal val loadedReporters: RegularFileProperty = objectFactory.fileProperty().convention(
+        projectLayout.intermediateResultsBuildDir("reporters.bin")
+    )
+
+    @get:OutputFile
+    internal val loadedReporterProviders: RegularFileProperty = objectFactory.fileProperty().convention(
+        projectLayout.intermediateResultsBuildDir("reporterProviders.bin")
+    )
+
+    @TaskAction
+    fun loadReporters() {
+        checkMinimalSupportedKtLintVersion(ktLintVersion.get())
+
+        val queue = workerExecutor.classLoaderIsolation { spec ->
+            spec.classpath.setFrom(ktLintClasspath, reportersClasspath)
+        }
```

Comment:
This doesn't do any dependency collision checking. So you could theoretically end up with multiple instances of, for example, guava here.

Reply:
Fixed via https://github.com/JLLeitschuh/ktlint-gradle/pull/424/commits/f8844a8faa4198e57ed24198cf947b5de67f03bf